File: app/src/test/java/com/permissionnanny/dagger/MockActivityComponent.java
package com.permissionnanny.dagger;

import com.permissionnanny.ConfirmRequestActivityTest;
import com.permissionnanny.missioncontrol.AppControlActivityTest;
import dagger.Component;

/**
 *
 */
@ContextComponent.ContextScope
@Component(modules = {ContextModule.class, ActivityModule.class}, dependencies = {AppComponent.class})
public interface MockActivityComponent extends ActivityComponent {

    void inject(AppControlActivityTest victim);

    void inject(ConfirmRequestActivityTest victim);
}

File: app/src/test/kotlin/com/permissionnanny/dagger/MockActivityComponent.kt
package com.permissionnanny.dagger

import com.permissionnanny.ConfirmRequestActivityTest
import com.permissionnanny.missioncontrol.AppControlActivityTest
import dagger.Component

/**

 */
@ContextComponent.ContextScope
@Component(modules = arrayOf(ContextModule::class, ActivityModule::class), dependencies = arrayOf(AppComponent::class))
interface MockActivityComponent : ActivityComponent {

    fun inject(victim: AppControlActivityTest)

    fun inject(victim: ConfirmRequestActivityTest)
}
